### PR TITLE
DOCS-2183 Stop bot protection for target URLs

### PR DESCRIPTION
**Pull request by MaximBashurov**

### CODE REVIEW

File: docs/latest/about-wallarm/api-abuse-prevention.md [hunk 1]
```diff
@@ -94,3 +94,11 @@ An exception list is a list of IP addresses, subnets, locations and source types
 You can add IP addresses to the exception list in advance or if they have already been mistakenly flagged as being associated with malicious bot activity. [Learn how to work with exception list →](../user-guides/api-abuse-prevention.md#working-with-exception-list)
 
 ![!API Abuse prevention - Exception list](../images/about-wallarm-waf/abi-abuse-prevention/exception-list.png)
+
+## Disabling bot protection for specific URLs
+
+Besides stopping bot protection for **source** URLs with [exception list](#exception-list), you can disable it for specific **target** URLs using the [**Set API Abuse mode**](../user-guides/rules/api-abuse-url.md) rule.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/api-abuse-prevention.md:100]: source URLs? check how exception lists work

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+![!Creating Set API Abuse mode for API Discovery endpoint](../../images/user-guides/rules/api-abuse-url.png)
+
+Note that you can temporarily disable the exception for the endpoint without deleting the rule by selecting the **Default** mode. You can go back to **Do not check for bot activity** at any moment later.
+
+### Set of URLs should not be checked for bot activities
+
+Let's say you need to stop checking bot activities for the set of URLs defined with the help of [regular expression](add-rule.md#condition-type-regex). Go to the **Rules** section and create the **Set API Abuse mode** rule:
+
+![!Creating Set API Abuse mode for set of URLs with regular expression](../../images/user-guides/rules/api-abuse-url-regex.png)
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:41]: image with grammarly

Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:41]: disable grammarly for such resource - we have discussed that

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Disabling bot protection for specific URLs
+
+The [**API Abuse Prevention**](../../about-wallarm/api-abuse-prevention.md) module of the Wallarm platform detects and mitigates bots. What applications are being protected, from bots of which types, and with which level of tolerance is configured by one or more [API abuse profiles](../../user-guides/api-abuse-prevention.md). Additionally, you can disable bot protection for specific URLs using the **Set API Abuse mode** rule described in this article.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:3]: bots of which types, with which level of tolerance - try to polish such constructions

Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:3]: and mitigates malicious bots

Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:3]: What applications are being protected, from bots of which types, and with which level of tolerance is configured by one or more [API abuse profiles](../../user-guides/api-abuse-prevention.md). - is extra

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+Note that this rule is different from the [exception list](../../about-wallarm/api-abuse-prevention.md#exception-list) of API Abuse Prevention: the exception lists sets **source** URLs, and the **Set API Abuse mode** rule - **target** URLs not to be checked.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:5]: the rule name is wrong. API Abuse mode does not exist

Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:5]: this sentence is extra. exception list and this rule are absolutely different things

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+To disable bot protection for specific URL:
+
+1. Proceed to Wallarm Console → **Rules** → **Add rule**.
+1. In **If request is**, [describe](add-rule.md#branch-description) the scope to apply the rule to.
+
+    If you use the [**API Discovery**](../../about-wallarm/api-discovery.md) module and have your endpoints discovered, you can also quickly create the rule for the endpoint using its menu.
+
+1. In **Then**, choose **Set API Abuse mode** and set:
+
+    * **Default** - for the described scope (specific URL or more), the protection from bots will work in a usual way defined by common API Abuse Prevention profiles.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:18]: the link to profile docs

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+Note that this rule is different from the [exception list](../../about-wallarm/api-abuse-prevention.md#exception-list) of API Abuse Prevention: the exception lists sets **source** URLs, and the **Set API Abuse mode** rule - **target** URLs not to be checked.
+
+## Creating and applying the rule
+
+To disable bot protection for specific URL:
+
+1. Proceed to Wallarm Console → **Rules** → **Add rule**.
+1. In **If request is**, [describe](add-rule.md#branch-description) the scope to apply the rule to.
+
+    If you use the [**API Discovery**](../../about-wallarm/api-discovery.md) module and have your endpoints discovered, you can also quickly create the rule for the endpoint using its menu.
```
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:14]: if the option is only available for this rule type, let's name the button correspondingly

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+1. In **Then**, choose **Set API Abuse mode** and set:
+
+    * **Default** - for the described scope (specific URL or more), the protection from bots will work in a usual way defined by common API Abuse Prevention profiles.
+    * **Do not check for bot activity** - for the described scope, the check for bot activity will not be performed.
+
+1. Optionally, in the comment, specify the reason of creating the rule for this URL.
+
+## Rule examples
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:23]: it is expected to get use cases in this section, not technical use cases when this rule is applied to an endpoint from an inventory or another one - this is worthy to mention in a real use case description just in one sentence. when users may need to disable malicious bot detection?

File: docs/latest/about-wallarm/api-abuse-prevention.md (hunk 1 above)
Comment (MaximBashurov) [reply docs/latest/about-wallarm/api-abuse-prevention.md:100]: > source URLs? exception list is for bot IPs

[Docs](https://docs.wallarm.com/user-guides/api-abuse-prevention/#working-with-exception-list): "legitimate bots or crawlers" + in the current context it is important that this is "where requests go from" as the opposite to "where requests go to".

File: docs/latest/user-guides/rules/api-abuse-url.md (hunk 3 above)
Comment (MaximBashurov) [reply docs/latest/user-guides/rules/api-abuse-url.md:3]: > extra

This has value - I want to oppose main config (profile) from additional (rule).

File: docs/latest/user-guides/rules/api-abuse-url.md (hunk 4 above)
Comment (MaximBashurov) [reply docs/latest/user-guides/rules/api-abuse-url.md:5]: > the rule name is wrong. API Abuse mode does not exist

The rule name is not bad. And if we agree on that, API Abuse mode will exist.

Comment (MaximBashurov) [reply docs/latest/user-guides/rules/api-abuse-url.md:5]: > exception list and this rule are absolutely different things

No, they are not. Both define URL exceptions, which is very similiar.

File: docs/latest/user-guides/rules/api-abuse-url.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/user-guides/rules/api-abuse-url.md:14]: > is this option available for all rule types?

We are saying that rule of THIS type can be created from the endpoint in API Discovery.

[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/user-guides/rules/api-abuse-url.md:14]: And rule of any type can be created from the APID endpoint page.

File: docs/latest/user-guides/rules/api-abuse-url.md (hunk 7 above)
Comment (MaximBashurov) [reply docs/latest/user-guides/rules/api-abuse-url.md:23]: We have "you have the endpoint that belongs to your application. The application should be protected from bot activities but the testing endpoint should be an exception".

Asked the team for more.

File: docs/latest/about-wallarm/api-abuse-prevention.md (hunk 1 above)
Comment (AnastasiaTWW) [reply docs/latest/about-wallarm/api-abuse-prevention.md]: URL is nit equal to IP

File: docs/latest/user-guides/rules/api-abuse-url.md (hunk 4 above)
Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:5]: ip is not url - it is important

File: docs/latest/user-guides/rules/api-abuse-url.md (hunk 7 above)
Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:23]: no. when a customer may need to disable protection for an endpoint? what use cases this feature covers? why endpoints may be excluded from the module?

File: docs/latest/user-guides/rules/api-abuse-url.md (hunk 4 above)
Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:5]: and the rule name is still wrong + specify the node version everywhere it is needed

File: docs/latest/about-wallarm/api-abuse-prevention.md [hunk 8]
```diff
@@ -94,3 +94,11 @@ An exception list is a list of IP addresses, subnets, locations and source types
...
 
 ![!API Abuse prevention - Exception list](../images/about-wallarm-waf/abi-abuse-prevention/exception-list.png)
+
+## Disabling bot protection for specific URLs and requests
+
+In addition to marking good bots' IPs via [exception list](#exception-list), you can disable bot protection both for URLs that the requests target and for the particular request types, for example, for the requests containing specific headers. You can do it using the [**Set API Abuse Prevention mode**](../user-guides/rules/api-abuse-url.md) rule.
+
+This may help to avoid false positive detections and may be also helpful in case of testing your applications when you may need to temporarily disable bot protection for some of their endpoints.
+
+Note that comparing to other API Abuse Prevention configuration, this ability is an additional one, thus it is configured [not within](../user-guides/api-abuse-prevention.md#working-with-exceptions-for-target-urls) the API Abuse profile, but separately - with the help of the rule.
```
Comment (AnastasiaTWW) [inline docs/latest/about-wallarm/api-abuse-prevention.md:104]: Note that comparing to other API Abuse Prevention configuration, this ability is configured NOT within the [API Abuse profile](../user-guides/api-abuse-prevention.md#working-with-exceptions-for-target-urls), but separately - with the help of the [rule].

shorter a little bit and the profile link is worng place

File: docs/latest/user-guides/api-abuse-prevention.md [hunk 9]
```diff
@@ -102,3 +102,9 @@ When the IP address is added to the exception list, the address is automatically
 By default, the IP is added to the exception list forever. You can change this and set time when the address should be removed from the exception list. You can also remove address from exceptions immediately at any moment.
 
 The **Exception list** tab provides the historical data - you can view items that were presented in the list within the selected period of time in past.
+
+## Working with exceptions for target URLs and specific requests
+
+In addition to marking good bots' IPs via [exception list](#working-with-exception-list), you can disable bot protection both for URLs that the requests target and for the particular request types, for example, for the requests containing specific headers. You can do it using the [**Set API Abuse Prevention mode**](../user-guides/rules/api-abuse-url.md) rule.
+
+Unlike other API Abuse Prevention configuration, this ability is an additional one, thus it is configured not within the API Abuse profile, but separately - with the help of the rule.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/api-abuse-prevention.md:110]: the same

Comment (AnastasiaTWW) [reply docs/latest/user-guides/api-abuse-prevention.md:110]: https://github.com/wallarm/product-docs-en/pull/928/files#r1289892839

File: mkdocs-4.6.yml [hunk 10]
```diff
@@ -270,6 +270,7 @@ nav:
           - Ignoring attack signs in the binary data: user-guides/rules/ignore-attacks-in-binary-data.md
           - The overlimit_res attack detection fine‑tuning: user-guides/rules/configure-overlimit-res-detection.md
           - Setting rate limit: user-guides/rules/rate-limiting.md
+          - Disabling bot protection for specific URLs and requests: user-guides/rules/api-abuse-url.md
```
Comment (AnastasiaTWW) [inline mkdocs-4.6.yml:273]: reflect the changes only in 4.8, 4.6 should not be affected

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    * **Default** - for the described scope (specific URL or request), the protection from bots will work in a usual way defined by common API Abuse Prevention [profiles](../../user-guides/api-abuse-prevention.md).
+    * **Do not check for bot activity** - for the described URL and/or request type, the check for bot activity will not be performed.
+
+1. Optionally, in the comment, specify the reason of creating the rule for this URL/request type.
+
+## Rule examples
+
+### Specific requests should not be checked for bot activities
+
+If you need to turn off a bot protection (mark a good bot) for a long period of time and do not want to manipulate IPs related to bots by placing or removing them from [exception list](../../about-wallarm/api-abuse-prevention.md#exception-list), you can do that by creating the **Set API Abuse Prevention mode** rule for specific request headers, and setting the mode to **Do not check for bot activity** so that such request will not get blocked.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:30]: this info is extra. you have already described the difference between the exception list and this rule above. this is obvious that the current article is about the rule, so it is normal just to skip this info

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+
+In this example we set not to check for bot activities in GET requests from the Klaviyo/1.0 user agent for specific URIs:
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:32]: read the examples in the rate limit rule. currently, your example is like: set the rule for specific header. it is an action not a use case. what is a use case?

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+Let's say you have your API inventory discovered the [**API Discovery**](../../about-wallarm/api-discovery.md) module and you have the endpoint that belongs to your application. The application should be protected from bot activities but the testing endpoint should be an exception.
+
+General protection is configured by API Abuse profile, exception for the endpoint is set via the **Set API Abuse Prevention mode** rule set to **Do not check for bot activity**.
+
+In this case it is easier to create rule from the **API Discovery** list of endpoints. Go there, find your endpoint and initiate rule creation from its page:
+
+![!Creating Set API Abuse Prevention mode for API Discovery endpoint](../../images/user-guides/rules/api-abuse-url.png)
+
+Note that you can temporarily disable the exception for the endpoint without deleting the rule by selecting the **Default** mode. You can go back to **Do not check for bot activity** at any moment later.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:48]: do not repeat it, move to the section about creating the rule

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+In this example we set not to check for bot activities in GET requests from the Klaviyo/1.0 user agent for specific URIs:
+
+![!Do not check for bot activity for requests with specific headers](../../images/user-guides/rules/api-abuse-url-request.png)
+
+Note that you can temporarily disable the exception for this type of requests without deleting the rule: to do that, select the **Default** mode. You can go back to **Do not check for bot activity** at any moment later.
+
+### Specific endpoint in your API inventory should not be checked for bot activities
+
+Let's say you have your API inventory discovered the [**API Discovery**](../../about-wallarm/api-discovery.md) module and you have the endpoint that belongs to your application. The application should be protected from bot activities but the testing endpoint should be an exception.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:40]: here is a good example of use case - disable protection for a testing endpoint

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+If you need to turn off a bot protection (mark a good bot) for a long period of time and do not want to manipulate IPs related to bots by placing or removing them from [exception list](../../about-wallarm/api-abuse-prevention.md#exception-list), you can do that by creating the **Set API Abuse Prevention mode** rule for specific request headers, and setting the mode to **Do not check for bot activity** so that such request will not get blocked.
+
+In this example we set not to check for bot activities in GET requests from the Klaviyo/1.0 user agent for specific URIs:
+
+![!Do not check for bot activity for requests with specific headers](../../images/user-guides/rules/api-abuse-url-request.png)
+
+Note that you can temporarily disable the exception for this type of requests without deleting the rule: to do that, select the **Default** mode. You can go back to **Do not check for bot activity** at any moment later.
+
+### Specific endpoint in your API inventory should not be checked for bot activities
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:38]: it is not a case. a case is disabling protection for a testing endpoint

File: docs/latest/user-guides/rules/api-abuse-url.md (hunk 14 above)
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:40]: start the paragraph from the use case. creating a rule for either the endpoint from the inventory if it was discovered or for the endpoint URL via the rule creation window is secondary. mention that here you describe the first method but in the 2nd sentence

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+Let's say you have your API inventory discovered the [**API Discovery**](../../about-wallarm/api-discovery.md) module and you have the endpoint that belongs to your application. The application should be protected from bot activities but the testing endpoint should be an exception.
+
+General protection is configured by API Abuse profile, exception for the endpoint is set via the **Set API Abuse Prevention mode** rule set to **Do not check for bot activity**.
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:42]: again it is extra, you have described that above

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+
+## Rule examples
+
+### Specific requests should not be checked for bot activities
+
+If you need to turn off a bot protection (mark a good bot) for a long period of time and do not want to manipulate IPs related to bots by placing or removing them from [exception list](../../about-wallarm/api-abuse-prevention.md#exception-list), you can do that by creating the **Set API Abuse Prevention mode** rule for specific request headers, and setting the mode to **Do not check for bot activity** so that such request will not get blocked.
+
+In this example we set not to check for bot activities in GET requests from the Klaviyo/1.0 user agent for specific URIs:
+
+![!Do not check for bot activity for requests with specific headers](../../images/user-guides/rules/api-abuse-url-request.png)
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:34]: the screenshot has empty state of the "Enter URI" field. but it cannot be empty

Comment (AnastasiaTWW) [reply docs/latest/user-guides/rules/api-abuse-url.md:34]: also, it has "/path" but this element is not reflected anywhere in the text

File: docs/latest/user-guides/rules/api-abuse-url.md [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+
+### Specific endpoint in your API inventory should not be checked for bot activities
+
+Let's say you have your API inventory discovered the [**API Discovery**](../../about-wallarm/api-discovery.md) module and you have the endpoint that belongs to your application. The application should be protected from bot activities but the testing endpoint should be an exception.
+
+General protection is configured by API Abuse profile, exception for the endpoint is set via the **Set API Abuse Prevention mode** rule set to **Do not check for bot activity**.
+
+In this case it is easier to create rule from the **API Discovery** list of endpoints. Go there, find your endpoint and initiate rule creation from its page:
+
+![!Creating Set API Abuse Prevention mode for API Discovery endpoint](../../images/user-guides/rules/api-abuse-url.png)
```
Comment (AnastasiaTWW) [inline docs/latest/user-guides/rules/api-abuse-url.md:46]: grammarly

File: mkdocs-4.6.yml (hunk 10 above)
Comment (MaximBashurov) [reply mkdocs-4.6.yml:273]: Is a part of future checklist.

Comment (MaximBashurov) [reply mkdocs-4.6.yml:273]: Is a part of future checklist.

**Pull request merged**